### ISSUE TITLE
kubernetes-client and kubernetes extension combinations don't generate the correct things

### ISSUE THREAD

**Issue by edeandrea** [quarkusio/quarkus#25688]
### Describe the bug

If I have a project which contains both the `kubernetes-client` extension and any of the other kubernetes extensions (`openshift`, `kubernetes`, `minikube`), if `src/main/kubernetes` contains other `Deployments`/etc, then the `ServiceAccount` object doesn't get generated, yet a `serviceAccountName` attribute gets added to each and every `Deployment` or `DeploymentConfig`.

 This should not happen.

### How to Reproduce?

Here's a reproducer:

[kubernetes-service-account.zip](https://github.com/quarkusio/quarkus/files/8731934/kubernetes-service-account.zip)

1. Download & unzip `kubernetes-service-account.zip`
2. `cd kubernetes-service-account`
3. `./mvnw clean package -DskipTests -Dquarkus.kubernetes.deployment-type=openshift`
4. Inspect `target/kubernetes/openshift.yml`

Notice there is is only a single `kind: ServiceAccount` object for the `fights-db`, yet the `DeploymentConfig` for `name: kubernetes-service-account` contains `serviceAccountName: kubernetes-service-account`, which does not exist.

This fails to deploy because the service account is missing.

**Comment by geoand** [comment]
@Sgitario you may be interested in this

**Comment by edeandrea** [comment]
FYI @michalszynkiewicz / @aureamunoz - I discovered this after I merged quarkusio/quarkus-super-heroes#82. After that merged I was no longer able to deploy the system to k8s using the yamls generated by the various kubernetes extensions.

I had to remove the `kubernetes-client` dependency and re-add the bouncycastle ones manually in order to work around this (see https://github.com/quarkusio/quarkus-super-heroes/blob/43ded01f2a077a61507526bde07447d0276caaa0/rest-fights/pom.xml#L171-L185). I'll revert that once this is fixed & rolled into a Quarkus release.
